packages/backend-common/src/reading/GithubUrlReader.test.ts

```diff
@@ -230,4 +243,91 @@ describe('GithubUrlReader', () => {
       );
     });
   });
+
+  describe('readTree', () => {
+    it('returns the files from a archive', async () => {
+      const worker = setupServer();
+
+      const repoBuffer = fs.readFileSync(
+        path.resolve(dirname, 'test-data', 'repo.tar.gz'),
+      );
+
+      worker.use(
+        rest.get(
+          'https://github.com/spotify/mock/archive/repo.tar.gz',
+          (_, res, ctx) =>
+            res(
+              ctx.status(200),
+              ctx.set('Content-Type', 'application/x-gzip'),
+              ctx.body(repoBuffer),
+            ),
+        ),
+      );
+      worker.listen({ onUnhandledRequest: 'error' });
+
+      const processor = new GithubUrlReader({
+        host: 'github.com',
+      });
+
+      const response = await processor.readTree(
+        'https://github.com/spotify/mock',
+        'repo',
+        ['mkdocs.yml', 'docs'],
+      );
+
+      const files = response.files();
+
+      const mkDocsFile = await files[0].content();
+      const indexMarkdownFile = await files[1].content();
+
+      expect(mkDocsFile.toString()).toBe('site_name: Test\n');
+      expect(indexMarkdownFile.toString()).toBe('# Test\n');
+    });
+
+    it('returns a folder path from a archive', async () => {
+      const worker = setupServer();
+
+      const repoBuffer = fs.readFileSync(
+        path.resolve(dirname, 'test-data', 'repo.tar.gz'),
+      );
+
+      worker.use(
+        rest.get(
+          'https://github.com/spotify/mock/archive/repo.tar.gz',
+          (_, res, ctx) =>
+            res(
+              ctx.status(200),
+              ctx.set('Content-Type', 'application/x-gzip'),
+              ctx.body(repoBuffer),
+            ),
+        ),
+      );
+      worker.listen({ onUnhandledRequest: 'error' });
+
+      const processor = new GithubUrlReader({
+        host: 'github.com',
+      });
+
+      const response = await processor.readTree(
+        'https://github.com/spotify/mock',
+        'repo',
+        ['mkdocs.yml', 'docs'],
+      );
+
+      mockfs();
+      const directory = await response.dir('/tmp/fs');
+
+      const writtenToDirectory = fs.existsSync(directory);
+      const paths = await recursive(directory);
+      mockfs.restore();
+
+      expect(writtenToDirectory).toBe(true);
+      expect(paths.sort()).toEqual(
+        [
+          '/tmp/fs/mock-repo/docs/index.md',
```

Comment:
Hm does this work on windows?

Reply:
Mock-fs library should handle the forward slashes in Windows. Not sure about the non windowsy path apart from that, but i assumed it would work. Aren't the tests run in windows as well? Can't see a workflow being run on this PR for it.

Reply:
yeah good point, maybe it's abstracted away here anyway